src/start/webAuth.js

```diff
@@ -71,19 +71,37 @@ export const authFromCallbackUrl = (
   otp: string,
   realm: string,
 ): Auth | null => {
-  const url: Url = parseURL(callbackUrl, true);
-
   // callback format expected: zulip://login?realm={}&email={}&otp_encrypted_api_key={}
+  const url = tryParseUrl(callbackUrl);
+
+  if (!url) {
+    return null;
+  }
```

Comment:
nit: for this sort of "check return value" pattern, no blank line -- the check and the line above it that produces the value are closely connected, so best to keep them visually together